src/FileFormats/LP/LP.jl

```diff
@@ -285,9 +285,475 @@ end
 #   Base.read!
 #
 # ==============================================================================
+const _COMMENT_REG = r"(.*?)\\(.*)"
+function _strip_comment(line::String)
+    if occursin("\\", line)
+        m = match(_COMMENT_REG, line)
+        return strip(String(m[1]))
+    else
+        return strip(line)
+    end
+end
+
+# a list of section keywords in lower-case
+const _KEYWORDS = Dict(
+    "max" => Val{:obj},
+    "maximize" => Val{:obj},
+    "maximise" => Val{:obj},
+    "maximum" => Val{:obj},
+    "min" => Val{:obj},
+    "minimize" => Val{:obj},
+    "minimise" => Val{:obj},
+    "minimum" => Val{:obj},
+    "subject to" => Val{:constraints},
+    "such that" => Val{:constraints},
+    "st" => Val{:constraints},
+    "s.t." => Val{:constraints},
+    "bounds" => Val{:bounds},
+    "bound" => Val{:bounds},
+    "gen" => Val{:integer},
+    "general" => Val{:integer},
+    "generals" => Val{:integer},
+    "bin" => Val{:binary},
+    "binary" => Val{:binary},
+    "binaries" => Val{:binary},
+    "end" => Val{:quit},
+)
+
+const _SENSE_ALIAS = Dict(
+    "max" => MOI.MAX_SENSE,
+    "maximize" => MOI.MAX_SENSE,
+    "maximise" => MOI.MAX_SENSE,
+    "maximum" => MOI.MAX_SENSE,
+    "min" => MOI.MIN_SENSE,
+    "minimize" => MOI.MIN_SENSE,
+    "minimise" => MOI.MIN_SENSE,
+    "minimum" => MOI.MIN_SENSE,
+)
+
+const _SUBJECT_TO_ALIAS = ["subject to", "such that", "st", "s.t."]
+
+const _CONSTRAINT_SENSE = Dict(
+    "<" => :le,
+    "<=" => :le,
+    "=" => :eq,
+    "==" => :eq,
+    ">" => :ge,
+    ">=" => :ge,
+)
+
+function _verify_name(variable::String, maximum_length::Int)
+    if length(variable) > maximum_length
+        return false
+    end
+    # m = match(START_REG, variable)
+    # if !isnothing(m)
+    #     return false
+    # end
```

Comment:
Is this commented for a reason?

Reply:
I forgot to fix this. I will look exactly what should be the STAR_REG because right now I think it blocks variables starting with `E` and I think it should allow them.

Reply:
I do love file formats with peculiar exceptions. Variables cannot start with an `E` or `e`:

![image](https://user-images.githubusercontent.com/8177701/149018343-499b49e4-aa09-4245-980f-e2c61a39ec7a.png)

Reply:
I agree that they should be allowed in this reader. I'd add a comment on the code though.
I wouldn't mind blocking `{E/e}-E[0-9]*` and `{E/e}-[0-9]*` if you guys want to.

Reply:
I guess we can be liberal for what we read, and conservative for what we write. So then I'm okay reading names starting with `E`. The writer should add a prefix to them.

Reply:
Agreed